### PR TITLE
Added prettier pre-commit hook

### LINKED ISSUE #2138: Support prettier formatting in pre-commit hook for json files
## Description of issue / requirement to address

JSON files aren't formatted when committed and can lead to noisy diff's when auto-formatted in some IDE's.

## Proposed solution

Standardise the formatting of JSON files using a Prettier pre-commit hook.

### PR DESCRIPTION
**Pull request by oliverfunk**
## Linked Issues

Closes #2138

## Description

Formats JSON files using the prettier pre-commit hook.

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by CoronelBuendia — APPROVED** [review]
Approving on the basis that all the changes are aesthetic in all the sub-modules I am responsible for. JSONs look much nicer, thanks!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -44,3 +44,9 @@ repos:
     - id: black-jupyter
       language_version: python3.8
       types: [jupyter]
+
+-   repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.5.1
+    hooks:
+    - id: prettier
+      types_or: [json]
```
Comment (je-cook) [inline .pre-commit-config.yaml:52]: Then we can remove the `.prettierrc` file
```suggestion
      types_or: [json]
      args: [--tab-width=2]
```

**Review by je-cook — CHANGES_REQUESTED** [review]
Just the one change please. Otherwise looks look

File: .pre-commit-config.yaml (hunk 1 above)
Comment (oliverfunk) [reply .pre-commit-config.yaml:52]: Ah that what it was!

**Review by je-cook — APPROVED** [review]

**Pull request merged**